test/cognito/index.test.js

```diff
@@ -0,0 +1,139 @@
+const {
+	MOCK_PROMISE,
+	mockUsername,
+	mockUsersPresent,
+	mockUserPoolId,
+} = require('../fixtures/aws/cognito/general.json');
+
+const {
+	mockUpdateUserPoolPayload,
+} = require('../fixtures/aws/cognito/mockEventPayload.json');
+
+const {
+	mockEmptyUsers
+} = require('../fixtures/aws/cognito/mockResponse.json');
+
+const mockPromiseFn = jest.fn(() => mockEmptyUsers);
+const mockPromise = {
+	promise: jest.fn(() => MOCK_PROMISE)
+};
+
+const mockUpdateUserPool = jest.fn(() => mockPromise);
+
+const mockCognitoIdentityServiceProvider = jest.fn(() => ({
+	updateUserPool: mockUpdateUserPool,
+}));
+
+const Cognito = require('../../utils/aws/cognito/index');
+
+jest.mock('aws-sdk', () => ({
+	CognitoIdentityServiceProvider: mockCognitoIdentityServiceProvider,
+}));
+
+beforeEach(() => {
+	jest.clearAllMocks();
+	jest.resetModules();
+});
+
+describe('Cognito - /utils/aws/cognito functonality', () => {
+	const {
+		AccountRecoverySetting,
+		AdminCreateUserConfig,
+		AutoVerifiedAttributes,
+		DeviceConfiguration,
+		EmailConfiguration,
+		EmailVerificationMessage,
+		EmailVerificationSubject,
+		LambdaConfig,
+		MfaConfiguration,
+		Policies,
+		SmsAuthenticationMessage,
+		SmsConfiguration,
+		SmsVerificationMessage,
+		UserPoolAddOns,
+		UserPoolId,
+		UserPoolTags,
+		VerificationMessageTemplate,
+	} = mockUpdateUserPoolPayload;
+
+	test('Should return async/await update userpool - updateUserPool()', async () => {
+		const updatePool = await Cognito.updateUserPool(
+			AccountRecoverySetting,
+			AdminCreateUserConfig,
+			AutoVerifiedAttributes,
+			DeviceConfiguration,
+			EmailConfiguration,
+			EmailVerificationMessage,
+			EmailVerificationSubject,
+			LambdaConfig,
+			MfaConfiguration,
+			Policies,
+			SmsAuthenticationMessage,
+			SmsConfiguration,
+			SmsVerificationMessage,
+			UserPoolAddOns,
+			UserPoolId,
+			UserPoolTags,
+			VerificationMessageTemplate,
+		);
+
+		expect(mockUpdateUserPool).toBeCalledTimes(1);
+		expect(mockUpdateUserPool).toBeCalledWith(mockUpdateUserPoolPayload);
+		expect(updatePool).toEqual(MOCK_PROMISE);
+	});
+
+	test('Should return resolves update userpool - updateUserPool()', async () => {
+		mockPromiseFn.mockResolvedValueOnce('updateUserPool - Success');
+
+		const updatePool = await Cognito.updateUserPool(
+			AccountRecoverySetting,
+			AdminCreateUserConfig,
+			AutoVerifiedAttributes,
+			DeviceConfiguration,
+			EmailConfiguration,
+			EmailVerificationMessage,
+			EmailVerificationSubject,
+			LambdaConfig,
+			MfaConfiguration,
+			Policies,
+			SmsAuthenticationMessage,
+			SmsConfiguration,
+			SmsVerificationMessage,
+			UserPoolAddOns,
+			UserPoolId,
+			UserPoolTags,
+			VerificationMessageTemplate,
+		);
+
+		expect(mockUpdateUserPool).toBeCalledTimes(1);
+		expect(updatePool).toEqual(MOCK_PROMISE);
+	});
+
+	test('Should return rejects update userpool - updateUserPool()', async () => {
+		mockPromiseFn.mockRejectedValueOnce(new Error('reject updateUserPool() thrown error'));
```

Comment:
Please check documentation for the correct error messages from Cognito and add them in the rejections and resolve